ClientApp/src/NavBar/NavBar.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState, useEffect, useRef, MouseEvent } from 'react';
+import './NavBar.scss';
+import { Link } from 'react-router-dom';
+import { Spin as Hamburger } from 'hamburger-react';
+
+function NavBar() {
+    const [selectedPage, setSelectedPage] = useState('null');
+    const [mobileNavState, setMobileNavState] = useState('closed')
+    const [isOpen, setOpen] = useState(false)
+
+    const menuRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+     let handler = (event: any) => {
+            if (menuRef.current !== null && !menuRef.current.contains(event.target)) {
+            setMobileNavState("closed");
+            setOpen(false);
+            }
+        };
+
+        document.addEventListener("mousedown", handler);
+
+        return () => {
+            document.removeEventListener("mousedown", handler)
+        }
```

Comment:
indentation in here looks a little off :)

Reply:
Fixed indentation :)